util/include/Metrics.hpp

```diff
@@ -29,8 +29,8 @@ class BasicGauge;
 template <class T>
 class BasicCounter;
 
-using Gauge = BasicGauge<uint64_t>;
-using Counter = BasicCounter<uint64_t>;
+using Gauge = BasicGauge<std::atomic_uint64_t>;
+using Counter = BasicCounter<std::atomic_uint64_t>;
```

Comment:
I think if we make all Counters/Gauges atomic, we might have a bit of performance degradation in the cases where they are only manipulated from a single thread. Even though by making this change we will solve all potential problems I think it might be better to change the types of the ones that are reported by TSAN to the atomic counterpart.

Reply:
@HristoStaykov : Both of them were reported by TSAN. That's why I have made changes in both.
<img width="894" alt="Screenshot 2021-06-02 at 3 29 32 PM" src="https://user-images.githubusercontent.com/12655005/120461140-61ab9b80-c3b7-11eb-8323-2b934579a7c6.png">

Reply:
Following were the Race conditions regarding Metric , so I have made them atomic accordingly

1. Write: BCStateTran.cpp:402: metrics_.last_block_.Get().Set(lastBlock);

Read : BCStateTran.cpp:647, metrics_component_.UpdateAggregator();

2. Write : BCStateTran.cpp:570 : metrics_.load_reserved_page_from_checkpoint_.Get().Inc();

Read : BCStateTran.cpp:647, metrics_component_.UpdateAggregator();

3. Write : BCStateTran.cpp:496 : metrics_.create_checkpoint_.Get().Inc();

Read : BCStateTran.cpp:647, metrics_component_.UpdateAggregator();

4. Write : BCStateTran.cpp:587, metrics_.save_reserved_page_.Get().Inc();

Read : BCStateTran.cpp:647, metrics_component_.UpdateAggregator();